### ISSUE TITLE
[instagram] Login with credentials is broken (`-u`/`-p` and `--netrc`)

### ISSUE THREAD

**Issue by Lyapsus** [yt-dlp/yt-dlp#7328]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.03.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Austria

### Provide a description that is worded well enough to be understood

I've got "An extractor error has occurred. (caused by KeyError('config'))" trying to download "https://www.instagram.com/stories/clublisicina/3125803082263395767/"

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://www.instagram.com/stories/clublisicina/3125803082263395767/', '--username', 'PRIVATE', '--password', 'PRIVATE']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d]
[debug] Python 3.11.4 (CPython x86_64 64bit) - Linux-6.3.8-1-cachyos-lto-x86_64-with-glibc2.37 (OpenSSL 3.1.1 30 May 2023, glibc 2.37)
[debug] exe versions: ffmpeg 6.0 (setts), ffprobe 6.0, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.12.0, brotlicffi-1.0.9.2, certifi-2023.05.07, mutagen-1.46.0, secretstorage-3.3.3, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1786 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
yt-dlp is up to date (stable@2023.03.04)
[instagram:story] Downloading login webpage
ERROR: 3125803082263395767: An extractor error has occurred. (caused by KeyError('config')); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 691, in extract
    self.initialize()
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 597, in initialize
    self._perform_login(username, password)
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/instagram.py", line 67, in _perform_login
    'X-CSRFToken': shared_data['config']['csrf_token'],
                   ~~~~~~~~~~~^^^^^^^^^^
KeyError: 'config'
```

**Comment by bashonly** [comment]
Same issue as #5154, but I think will close that in favor of this because of the signal:noise ratio in the older issue

### TL;DR:
 - instagram login w/ `-u`/`-p` or `--netrc` is broken
 - PR #6294 is an attempt to fix this that may also be broken
 - you can try using `--cookies-from-browser` or `--cookies` instead
 - **disclaimer**: using an IG account with yt-dlp is risky. IG employs various methods of detecting automated tool usage, and they do not hesitate to permanently ban accounts when detected

**Comment by Lyapsus** [comment]
Greatly sorry, was looking for duplicates with downloading issue in mind, not actual login issue.
—cookies and —cookies-from-browser did the job even though there is no thorium profile (did “chromium:~/.config/thorium”)

**Comment by nikalasmd** [comment]
[Instagram] Downloading login webpage ERROR: C_uqvECRSSp: An extractor error has occurred. (caused by KeyError('config'));

[debug] Command-line config: ['-vU', '-u', 'PRIVATE', '-p', 'PRIVATE', 'https://www.instagram.com/reel/C_uqvECRSSp/']
[debug] Encodings: locale cp1251, fs utf-8, pref cp1251, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.08.06 from yt-dlp/yt-dlp [4d9231208] (pip)
[debug] Python 3.11.1 (CPython AMD64 64bit) - Windows-10-10.0.22000-SP0 (OpenSSL 1.1.1q  5 Jul 2022)
[debug] exe versions: none
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2023.07.22, mutagen-1.47.0, requests-2.32.3, sqlite3-3.39.4, urllib3-2.0.5, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1830 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.08.06 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.08.06 from yt-dlp/yt-dlp)
[Instagram] Downloading login webpage
ERROR: C_uqvECRSSp: An extractor error has occurred. (caused by KeyError('config')); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "C:\Users\Nicu\AppData\Local\Programs\Python\Python311\Lib\site-packages\yt_dlp\extractor\common.py", line 737, in extract
    self.initialize()
  File "C:\Users\Nicu\AppData\Local\Programs\Python\Python311\Lib\site-packages\yt_dlp\extractor\common.py", line 643, in initialize
    self._perform_login(username, password)
  File "C:\Users\Nicu\AppData\Local\Programs\Python\Python311\Lib\site-packages\yt_dlp\extractor\instagram.py", line 68, in _perform_login
    'X-CSRFToken': shared_data['config']['csrf_token'],
                   ~~~~~~~~~~~^^^^^^^^^^
KeyError: 'config'
